rest_live/consumers.py

```diff
@@ -122,23 +123,20 @@ def receive_json(self, content: Dict[str, Any], **kwargs):
                     view.request, view
                 )
 
-            # Retrieve actions must check has_object_permission as well.
+            # Retrieve actions use get_object() to check object permissions as well.
             if view.action == "retrieve":
+                view.kwargs.setdefault(view.lookup_field, lookup_value)
                 try:
-                    instance = view.get_queryset().get(
-                        **{view.lookup_field: lookup_value}
+                    view.get_object()
```

Comment:
Calling `get_object()` automatically applies filters and also checks permissions, which simplifies the logic in the consumer